lib/resque/pool.rb

```diff
@@ -344,7 +344,42 @@ def quit_excess_workers_for(queues)
     end
 
     def worker_delta_for(queues)
-      config.fetch(queues, 0) - workers.fetch(queues, []).size
+      config_get_worker_count(queues).to_i - workers.fetch(queues, []).size
+    end
+
+    # Get the number of workers declared in the config file for queues
+    #
+    # @param queues [String] names of queues separated by comma if many exist
+    def config_get_worker_count(queues)
+      worker_count = nil
```

Comment:
The thing is that in specs if a queue is not found (simply because an environment is set and the queue is not part of that) then nil is returned for the number of workers. This is why i kept this behavior and i think it makes sense

Reply:
@0xdiba ignore the previous comment. This method will return an integer only, i will fix this tomorrow.

Reply:
Fixed!